src/main/java/io/openliberty/sample/application/CrewService.java

```diff
@@ -35,26 +35,22 @@ public class CrewService {
 	@Inject
 	CrewMembers crewMembers;
 
-	@Inject
-	Validator validator;
-
 	@POST
 	@Path("/{id}") 
 	@Consumes(MediaType.APPLICATION_JSON)
 	@Produces(MediaType.APPLICATION_JSON) 
 	public String add(CrewMember crewMember) {
-		
-		Set<ConstraintViolation<CrewMember>> violations = validator.validate(crewMember);
-		if(violations.size() > 0) {
+
+		try {
+			crewMembers.save(crewMember);
+		} catch (ConstraintViolationException x) {
 			JsonArrayBuilder messages = Json.createArrayBuilder();
-			for (ConstraintViolation<CrewMember> v : violations) { 			
+			for (ConstraintViolation<?> v : x.getConstraintViolations()) {
```

Comment:
I had tried this before, but I was missing this piece. I didn't realize we could get the violations out of the exception, so we can still return them to the UI

Reply:
Actually I think I was validating at the REST layer, instead of with jakarta data.